content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,520 @@
+ 
+However, interpreting any form of t-test largely depends on the p-value, the result above explains that the means of the two samples(petal_width and sepal_length) are different. Thus, we will reject the Null Hypothesis which says the mean of the two samples are the same because their variances are assumed to be equal, there is no evidence sustainable enough to prove the two populations have the same mean.
+H1 (alternate hypothesis) is however true.
+ 
+### ANOVA - Python implementation
+ 
+ANOVA is a word coined from 'Analysis of Variance', it is a statistical concept that shows the differences between the means of more than two independent classes, using variance analysis on samples from those classes.
+It is used to check the contrast between three or more samples with one test. Especially when the categorical class has more than two categories.
+ 
+Why ANOVA? With ANOVA you get to discover obvious differences between the means of your independent features, and on getting a clearer picture of the differences, you can get to understand how each of them connects to your dependent variable, and what are the influencing factors for the relationship.
+ 
+A typical example:
+'The growth department of a certain company discovered an increase in sales for a particular product. In recent times, they've tried several advertisement channels so it is difficult to say which one yielded the sales increase the most.’
+ 
+They did online posters, billboards, TV ads, mouth-to-mouth, and sponsorship. By carrying out an ANOVA test on the several advertisement channels, they will be able to better understand which one works best out of all channels(independent variables), with the connection it has with the increase in sales which is the dependent variable.
+This can be applied in the marketing department of every organization.
+ 
+During ANOVA testing, the hypothesis is:
+ 
+**H0:** When all samples' means are the same.
+ 
+**H1:** When one or more samples are very much different.
+ 
+#### One Way ANOVA Test
+```python
+class1 = df.petal_width
+class2 = df.sepal_length
+class3 = df.petal_length
+```
+```python
+f_oneway(class1, class2, class3)
+```
+```bash
+F_onewayResult(statistic=555.8702565323317, pvalue=5.738282463819433e-122)
+```
+ 
+Since our P-value (5.738282463819433e-122) is far less than alpha = 0.05, therefore, we reject the Null Hypothesis, there is no evidence sustainable enough to accept it.
+ 
+This means that the sample means are very much different. Hence, our H1 (alternate hypothesis) is thus true.
+ 
+#### Two Way ANOVA Test
+ 
+This is called for when we are dealing with three or more variables, trying to compare their means with each other.
+```python
+petal = df.petal_width
+sepal = df.sepal_length
+species = df.species
+ 
+model = ols('species ~ C(petal) + C(sepal) + C(petal):C(sepal)', data=df).fit()
+sm.stats.anova_lm(model, typ=2)
+```
```

Comment:
This block of error seems to produce some error. Do look into this..
This is the error I'm getting when I ran it in my machine - 
```bash
ValueError: endog has evaluated to an array with multiple columns that has shape (150, 3). This occurs when the variable converted to endog is non-numeric (e.g., bool or str).
```
And do add the print statement.

Reply:
i'm afraid the error is from your end, i am not having  a similar error here...probably some dependencies are missing on your system, try use kaggle environment or google colab.